src/V128.cs

```diff
@@ -92,7 +92,14 @@ public Span<byte> AsSpan()
         {
             unsafe
             {
+#if NETSTANDARD2_0
+                fixed (byte* bytesPtr = bytes)
+                {
+                    return new Span<byte>(bytesPtr, 16);
+                }
```

Comment:
Note that on platforms without built-in `Span` support like .NET Framework, this will only be valid if the `V128` is on the stack, because the GC won't be able to track a reference to an object field.

For example, with a code like the following:
```cs
class MyV128Container
{
    public V128 v128;
    ~MyV128Container() => Console.WriteLine("Finalized MyV128Container.");
}

Span<byte> GetV128Span()
{
    var boxedV128 = new MyV128Container();
    return boxedV128.v128.AsSpan();
}

var span = GetV128Span();
GC.Collect();
GC.WaitForPendingFinalizers();
byte b1 = span[0];
```
When running on .NET 7.0 and setting a breakpoint on the `byte b1` line, the finalizer will not have run, as the `Span<byte>` tracks a field of the `MyV128Container` object, and so it is valid to access the `Span` afterwards.
However, when running it on .NET Framework, the finalizer will have run, as there the `Span<byte>` just stores a pointer value, and so accessing the `Span` outside of the `fixed` block would be invalid.

I'm not sure if there is a better solution for this (changing `V128` to a `ref struct` would cause e.g. the `V128.AllBitsSet` field to no longer work).
However, even on .NET 7.0 a similar issue can happen, when you declare a local `V128` variable (so it's on the stack), call `.AsSpan()` and return the span, as then the caller will get a span that would point to a stack area that is no longer valid.

Reply:
> In that case, I think it would be best to just replace the usage of `GetSpan` with `fixed` where appropriate, and remove `GetSpan` from the .NET Standard 2.0 version altogether ([fcaa87f](https://github.com/bytecodealliance/wasmtime-dotnet/commit/fcaa87f7994b0234b1f6281c90fed8a4a5b345ef)).

👍 
I think it's OK to not expose a `AsSpan()` method for .NET Standard 2.0 for now. The `V128` will still be usable by using one of its constructors to set its value, and by using the `.CopyTo(Span<byte>)` method to access its value.




> The best solution is, I think, to make the reference explicit through an extension method like this one:

Good idea! With that variant, this will actually trigger the `CS8347` error in the example (I didn't know that it's possible to declare an extension method that has a `ref this` parameter, and then call it without actually specifying the `ref` keyword):

![grafik](https://user-images.githubusercontent.com/13289184/237017890-1bee4e92-9cfe-4b3f-9485-0dd6674c45f4.png)

Reply:
Yup, it was added relatively recently (only for value types), alongside `in` parameters I think. 😉